ci: Upgrade the VSCode extension to 0.12.2 (#531)
PRQL/prql-vscode: DEPLOYING.md

Reviewer: Given the repo itself is mostly for the developer of the extenstion — there's a page on the VS Code marketplace — I would have it all in the Readme. But I defer if you feel strongly!

Author: Good point. I'll merge them back together

# Deploying and Development

The VSCode Extension for PRQL relies on the `prqlc` code from the
main PRQL repo. There are two Github Actions that wacth for updates
in this directory to build and release the extension to the
Microsoft Marketplace.

## Deploying/Updating the Extension

> When there is a new version of PRQL (`prqlc` in particular):

Reviewer: ```suggestion
When there is a new version of `prqlc` in `npm`, dependabot will PR an update.

Once that's merged:
```

> - In _package.json_, update:
> 
>   - `version` to the proper version (from the main PRQL repo)
>   - `dependencies.prqlc` to the same version
>   - (optional) update other dependencies and devDependencies

Reviewer: ```suggestion
- In [package.json](package.json), update `version` to the matching version from the main PRQL repo.
```

Author: There are actually two `version`s that I assume should be the same. True?
- right near the top of the JSON object (for the extension's version itself, I imagine)
- in `dependencies`, there's a `prqlc` property that tells which version to include

> - Check _.github/workflows/pull-request.yaml_ and _.github/workflows/release.yml_ to ensure that the `node-version` is correct.

Reviewer: I don't think this is necessary? We have to do it occasionally but it's not in the release checklist (what does "correct" mean here?)

> - Run `npm install` to get/update dependencies
> 
> - Run `npm run compile`. to build the extensions files
> 
> - (I AM GUESSING HERE...) The `.github` workflows check the
>   `main` branch on a daily basis to see whether there have been
>   changes and push a new extension as needed.

Reviewer: ```suggestion
- Run `npm install` to update the `package-lock.json`
- Create a new release. This will start a workflow to release the current version to the VS Code Marketplace
```

Author: I don't know how to 'create a new release' - that's part of the reason I want to document it here...
